src/utils.ts

```diff
@@ -25,4 +25,24 @@ const AsyncFunctionConstructor = (async () => {}).constructor;
 /**
  * an async function check method only consider runtime support async syntax
  */
-export const isAsyncFunction = (fn: Fn) => fn.constructor === AsyncFunctionConstructor;
+export const isAsyncFunction = async (task: Task) => {
+  if (task.fn.constructor === AsyncFunctionConstructor) {
```

Comment:
isn't this function only enough?

Reply:
I guess with the new condition it determines functions that return promises? right?

Reply:
Ok, I get it now. It's reasonable.